src/vos/tests/vts_io.c

```diff
@@ -2444,6 +2444,98 @@ io_query_key(void **state)
 	assert_int_equal(rc, -DER_NONEXIST);
 }
 
+static void
+update_dkey(void **state, daos_unit_oid_t oid, daos_epoch_t epoch,
+	    uint64_t dkey_value, const char *val)
+{
+	struct io_test_args	*arg = *state;
+	daos_iod_t		iod = {0};
+	d_sg_list_t		sgl = {0};
+	daos_key_t		dkey;
+	daos_key_t		akey;
+	d_iov_t			val_iov;
+	daos_recx_t		recx;
+	uint64_t		akey_value = 0;
+	int			rc = 0;
+
+	d_iov_set(&dkey, &dkey_value, sizeof(dkey_value));
+	d_iov_set(&akey, &akey_value, sizeof(akey_value));
+
+	iod.iod_type = DAOS_IOD_ARRAY;
+	iod.iod_name = akey;
+	iod.iod_recxs = &recx;
+	iod.iod_nr = 1;
+
+	/* Attach buffer to sgl */
+	d_iov_set(&val_iov, &val, strnlen(val, 32) + 1);
+	sgl.sg_iovs = &val_iov;
+	sgl.sg_nr = 1;
+
+	iod.iod_size = 1;
+	/* Set up rexs */
+	recx.rx_idx = 0;
+	recx.rx_nr = val_iov.iov_len;
+
+	rc = vos_obj_update(arg->ctx.tc_co_hdl, oid, epoch++, 0,
```

Comment:
Don't think '++' on epoch is needed here ... probably copy/paste error?

Doesn't have any ill effect so I'm okay with it getting fixed in a future patch.

Reply:
yeah, you're right.   I'll clean it up in another patch.